pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2021 ScyllaDB
+
+package operator
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+)
+
+func TestMain(m *testing.M) {
+	dynamiccertificates.FileRefreshDuration = 1 * time.Second
+	os.Exit(m.Run())
+}
+
+func TestRunWebhookServerCertReload(t *testing.T) {
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir := t.TempDir()
+	TLSCertFile := dir + "/tls.crt"
+	TLSKeyFile := dir + "/tls.key"
+
+	cc, err := newCertificateCreator()
+	if err != nil {
+		t.Fatalf("can't create CertificateCreator: %v", err)
+	}
+
+	certSerialNumber := big.NewInt(1)
+	encodedCert, err := cc.newEncodedCertificate(certSerialNumber)
+	if err != nil {
+		t.Fatalf("can't create new encoded certificate: %v", err)
+	}
+	if err = createFileWithContent(TLSCertFile, encodedCert.cert); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+	if err = createFileWithContent(TLSKeyFile, encodedCert.privateKey); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+
+	wo := WebhookOptions{
+		TLSCertFile:                    TLSCertFile,
+		TLSKeyFile:                     TLSKeyFile,
+		Port:                           5000,
+		InsecureGenerateLocalhostCerts: false,
+	}
+	err = wo.Complete()
+	if err != nil {
+		t.Fatalf("can't complete WebhookOptions: %v", err)
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		err := wo.run(ctx, genericclioptions.IOStreams{})
+		if err != nil {
+			t.Errorf("can't run webhook server: %v", err)
+		}
+	}()
+
+	encodedCa, err := cc.getEncodedCA()
+	if err != nil {
+		t.Fatalf("can't get encoded CA: %v", err)
+	}
+
+	certpool := x509.NewCertPool()
+	certpool.AppendCertsFromPEM(encodedCa)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: certpool,
+			},
+		},
+	}
+
+	addr := fmt.Sprintf("https://localhost:%d/readyz", wo.Port)
+
+	err = wait.PollImmediate(1*time.Second, 15*time.Second, func() (bool, error) {
+		resp, err := client.Get(addr)
+		return err == nil && reflect.DeepEqual(resp.TLS.PeerCertificates[0].SerialNumber, certSerialNumber), nil
+	})
+	if err != nil {
+		t.Fatalf("can't poll immediatelly: %v", err)
```

Comment:
`failed to observe the initial certificate: %v` (with fatal errors you don't always have to  [but can] use the regular wording as in the operator where it needs to imply it will retry)